spring-batch-integration/src/main/java/org/springframework/batch/integration/partition/RemotePartitioningMasterStepBuilder.java

```diff
@@ -0,0 +1,297 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.batch.integration.partition;
+
+import org.springframework.batch.core.Step;
+import org.springframework.batch.core.StepExecutionListener;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.batch.core.partition.PartitionHandler;
+import org.springframework.batch.core.partition.StepExecutionSplitter;
+import org.springframework.batch.core.partition.support.Partitioner;
+import org.springframework.batch.core.partition.support.StepExecutionAggregator;
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.batch.core.step.builder.PartitionStepBuilder;
+import org.springframework.batch.core.step.builder.StepBuilder;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.core.MessagingTemplate;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.StandardIntegrationFlow;
+import org.springframework.integration.dsl.context.IntegrationFlowContext;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.util.Assert;
+
+/**
+ * Builder for a master step in a remote partitioning setup. This builder creates and
+ * sets a {@link MessageChannelPartitionHandler} on the master step.
+ *
+ * <p>If no <code>messagingTemplate</code> is provided through
+ * {@link RemotePartitioningMasterStepBuilder#messagingTemplate(MessagingTemplate)},
+ * this builder will create one. The <code>outputChannel</code> set with
+ * {@link RemotePartitioningMasterStepBuilder#outputChannel(MessageChannel)} will be
+ * used as a default channel of the messaging template and will override any default
+ * channel already set on the messaging template.
+ *
+ * @since 4.1
+ * @author Mahmoud Ben Hassine
+ */
+public class RemotePartitioningMasterStepBuilder extends PartitionStepBuilder {
+
+	private static final long DEFAULT_POLL_INTERVAL = 10000L;
+	private static final long DEFAULT_TIMEOUT = 3600000L;
+
+	private MessagingTemplate messagingTemplate;
+	private MessageChannel inputChannel;
+	private MessageChannel outputChannel;
+	private JobExplorer jobExplorer;
+	private BeanFactory beanFactory;
+	private long pollInterval = DEFAULT_POLL_INTERVAL;
+	private long timeout = DEFAULT_TIMEOUT;
+
```

Comment:
Why are the required values that are provided by the factory (`JobExplorer`, etc) not configured via the constructor?

Reply:
To be consistent with other builders.